### PR TITLE
School library: decorate a class

### PR DESCRIPTION
**Pull request by username-forGithub**
- Implemented composition (as an example of the design pattern), using decorator and nameable class.
- Run a program using the command line.
- Used the Decorator design pattern to correct the names of people.
- use two decorators in order to capitalize and trim people's names.
- Implement a method called correct_name that will raise a NotImplementedError
- Implement the correct_name method that returns the result of the correct_name method of the @nameable

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (TedLivist) [inline nameable.rb:18]: - [ ] **[OPTIONAL]** Based on Ruby conventions, I would recommend you put this class in its own file

File: nameable.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (TedLivist) [inline nameable.rb:24]: - [ ] **[OPTIONAL]** Based on Ruby conventions, I would recommend you put this class in its own file

File: nameable.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
+  end
+end
```
Comment (TedLivist) [inline nameable.rb:30]: - [ ] **[OPTIONAL]** Based on Ruby conventions, I would recommend you put this class in its own file.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @username-forGithub,

✔️ Implementation of inheritance and class decoration

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: nameable.rb (hunk 3 above)
Comment (username-forGithub) [reply nameable.rb:30]: @TedLivist Thank you.

**Pull request merged**